whitelist-authenticator/src/main/java/de/intension/authentication/WhitelistAuthenticatorFactory.java

```diff
@@ -0,0 +1,83 @@
+package de.intension.authentication;
+
+import org.keycloak.Config;
+import org.keycloak.authentication.Authenticator;
+import org.keycloak.authentication.AuthenticatorFactory;
+import org.keycloak.models.AuthenticationExecutionModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.provider.ProviderConfigProperty;
+
+import java.util.List;
+
+/**
+ * Check KC_IDP_HINT against a configured whitelist.
```

Comment:
this class is basically a factory class to configure the WhitelistAuthenticator, the actual check happens in the WhitelistAuthenticator class

Reply:
your right, i changed it accordingly.